sklearnex/utils/validation.py

```diff
@@ -14,4 +14,162 @@
 # limitations under the License.
 # ===============================================================================
 
-from daal4py.sklearn.utils.validation import _assert_all_finite
+import numbers
+
+import scipy.sparse as sp
+from sklearn.utils.validation import _assert_all_finite as _sklearn_assert_all_finite
+from sklearn.utils.validation import _num_samples, check_array, check_non_negative
+
+from daal4py.sklearn._utils import daal_check_version, sklearn_check_version
+
+from ._array_api import get_namespace
+
+if sklearn_check_version("1.6"):
+    from sklearn.utils.validation import validate_data as _sklearn_validate_data
+
+    _finite_keyword = "ensure_all_finite"
+
+else:
+    from sklearn.base import BaseEstimator
+
+    _sklearn_validate_data = BaseEstimator._validate_data
+    _finite_keyword = "force_all_finite"
+
+
+if daal_check_version((2024, "P", 700)):
+    from onedal.utils.validation import _assert_all_finite as _onedal_assert_all_finite

```

Comment:
Can we add assert_all_finite to onedal.utils init.py? Feel free to use discretion but from my perspective it makes sense to 1. add to init, 2. make assert_all_finite accessible (instead of _assert_all_finite) or otherwise just delete assert_all_finite if its not being used, but you're deeper in the weeds here so let me know what you think makes sense

Reply:
And to go further on this - would it make sense to `from sklearn.utils import assert_all_finite` instead of `from sklearn.utils.validation import _assert_all_finite` ? Seems like this is the intended use, but maybe there is overhead with this option, any thoughts?

Reply:
The underlying ```_assert_all_finite``` is used as ```assert_all_finite``` is a thin wrapper for accessing sparse data when necessary. It is best only to that check once for sparsity at the initial use of ```assert_all_finite```. The original inclusion was made in onedal in order to verify sparse support, and the testing was converted to ```assert_all_finite``` because of this: https://github.com/uxlfoundation/scikit-learn-intelex/pull/2126#discussion_r1861345654 . We can delete the sparse conversion version from onedal with the expectation that we will have to convert the testing there as well.

Reply:
Thanks for clarification, let's leave onedal4py as is and continue to use _assert_all_finite in cases we know are not sparse then